equational_theories/ManuallyProved/Equation1729/SmallMagma.lean

```diff
@@ -97,12 +106,26 @@ lemma SM_obeys_1729 : Equation1729 SM := by
   abel_nf
   exact SM_char_four _
 
-lemma E_ne_SE (n m : ℕ): E n ≠ S (E m) := by
+lemma one_odd (n : ZMod 4) : 1 ≠ n + n := by
+  revert n
+  decide
+
+lemma E_ne_S (n:ℕ) (a:SM) : E n ≠ S a := by
   by_contra! this
   apply_fun (fun f ↦ f n) at this
-  by_cases h:m=n
-  all_goals simp [E,S, DirectSum.of_apply,h] at this
-  all_goals contrapose! this; decide
+  simp [E,SM_square_eq_double, DirectSum.of_apply] at this
+  exact one_odd (a n) this
```

Comment:
It can be simplified as follows:

```suggestion
  simp [E,SM_square_eq_double, DirectSum.of_apply] at this
  exact one_odd (a n) this
```

Tested locally so it's safe to commit directly.